libs/UIHelpers/histogram.cpp

```diff
@@ -0,0 +1,105 @@
+#include "histogram.h"
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <iomanip>
+#include <sstream>
+
+#include "Vec2.h"
+
+template <typename T>
+T clamp(T x, T min, T max)
+{
+    if (x < min) {
+        return min;
+    }
+    if (x > max) {
+        return max;
+    }
+    return x;
+}
+
+Histogram::Histogram()
+{
+  m_min_max_val[0] = 0.0f;
+  m_min_max_val[1] = 1.0f;
+}
+
+
+void Histogram::setHistogram(const std::vector<float> &hist)
+{
+	current_histogram = hist;
+}
+
+std::vector<float>& Histogram::getHistogram()
+{
+  return current_histogram;
+}
+
+void Histogram::setMinMax(const float min, const float max) {
+	m_min_max_val[0] = min;
+	m_min_max_val[1] = max;
+}
+
+
+void Histogram::setBlendedHistogram(const std::vector<float>& hist1, const std::vector<float>& hist2, float alpha)
+{
+	if (hist1.size() != hist2.size())
+		return;
+
+	current_histogram.clear();
+	for (int i = 0; i < hist1.size(); i++)
+		current_histogram.push_back(hist1[i] * alpha + hist2[i] * (1.0f - alpha));
+}
+
+
+void Histogram::draw_histogram()
```

Comment:
I think ImGUI has `ImGui::PlotHistogram` method. Is that one not useful?

Reply:
I just realize the  ImGui::PlotHistogram  widget exists !!!!!!
I am following Ben's implementation that had the histogram embedded in the transfer function widget. I think the one we have is useful enough.